### PR TITLE
Documentation Amendment: Updated description for authenticateOAuth2 security directive

### PR DESCRIPTION
**Pull request by trycatchkamal**
## Purpose

Documentation Amendment

## Changes

Updated description for authenticateOAuth2 security directive

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @trycatchkamal,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by raboof** [comment]
OK TO TEST

File: docs/src/main/paradox/routing-dsl/directives/security-directives/authenticateOAuth2.md [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 ## Description
 
-Wraps the inner route with OAuth Bearer Token authentication support using a given @scala[`AuthenticatorPF[T]`]@java[`AuthenticatorPF<T>`].
+Wraps the inner route with OAuth Bearer Token authentication support using a given @scala[`Authenticator[T]`]@java[`Authenticator<T>`].
```
Comment (raboof) [inline docs/src/main/paradox/routing-dsl/directives/security-directives/authenticateOAuth2.md]: For the Java API It looks like this directive takes a function from `Optional<ProvidedCredentials>` to `Optional<T>`.

For the Scala variation, would it be possible to use the `@scaladoc` directive to link to https://doc.akka.io/api/akka-http/current/akka/http/scaladsl/server/Directives$.html#Authenticator[T]=akka.http.scaladsl.server.directives.Credentials=%3EOption[T] ?

Comment (trycatchkamal) [reply docs/src/main/paradox/routing-dsl/directives/security-directives/authenticateOAuth2.md]: sure, will do

Comment (trycatchkamal) [reply docs/src/main/paradox/routing-dsl/directives/security-directives/authenticateOAuth2.md]: Lot more improvements done in https://github.com/akka/akka-http/pull/2518 as suggested, thanks!

**Pull request merged**